Substitui a autenticação dos cookies por token

WellesleyMussolini opened the pull request.

WellesleyMussolini committed:
refactor: reorganize authentication cookies and constants folder structure

- Moved authentication cookies from an outdated constants folder to a more organized location.
- The new location is src/app/common/constants, part of the src/app/common directory.
- This restructuring enhances organization by centralizing shared applications.
- The common folder now houses all shared resources, making the project structure cleaner and more intuitive.

WellesleyMussolini committed:
refactor: reorganize protected routes constants folder structure

Moved the protectedRoutes array, which defines routes restricted to non-privileged users, from an outdated constants location to a more organized directory.

WellesleyMussolini committed:
refactor: update and correct Pathnames enum

- Added the missing /cart route to the enum.
- Corrected an incorrectly named route written by Matheus.

WellesleyMussolini committed:
refactor: move authentication cookies to common constants folder

I relocated the authentication cookie logic to the src/app/common/constants path.

WellesleyMussolini committed:
chore: update authentication cookie import path

WellesleyMussolini committed:
chore: update authentication cookie import path

WellesleyMussolini committed:
chore: temporary update to import in cart summary component

- Adjusted the import name from PrimaryButtonStyles to prevent a code-breaking issue in the cart summary component.
- The import will be correctly renamed to PrimaryButtonTypes in a future refactor, as this branch is currently focused on authentication-related updates.

WellesleyMussolini committed:
chore: Apply temporary changes

- Updated the import of IBoardGame to BoardGame to simplify naming and avoid redundancy in type definitions.
- Applied a temporary change to the import to prevent code-breaking issues, with plans for a future refactor to address the naming convention.
- This branch is focused on authentication, so further changes will be made in a separate refactor.

WellesleyMussolini committed:
refactor: relocate auth options to dedicated (auth) folder and improve session handling

The uthOptions function was moved from the global utils folder to the utils folder inside the uth folder, located in src/app/(auth).
This change simplifies understanding and managing the authentication structure, making debugging and further modifications easier.

In addition to the relocation, updates were made to the uthOptions. The session now uses the strategy: 'jwt', enabling JWT-based sessions. A new jwt method was introduced to set the token’s role based on the user session. The method checks if a user session exists, and if so, assigns the token's role to the user's role before returning the token.

I also commented out the previous cookie logic in the session method and now directly return the session values necessary for the application. The session secret is now provided by the .env file, referencing the OAuth secret for Google authentication. This modification allows the getToken method from NextAuth to retrieve the user session directly in the middleware, removing the need for a complex cookie logic that previously set the user’s role.

These changes not only reduce redundant code but also simplify the process of checking user roles and blocking routes based on authentication status, streamlining the overall logic.

WellesleyMussolini committed:
refactor: optimize middleware by removing cookie-based checks for authentication

With the recent changes in the uthOptions, the middleware no longer relies on cookies to check the user's authentication status or role. Now, the getToken function retrieves the session directly, allowing us to access the user's role and authentication status without the need for the previous cookie logic.

This refactor significantly reduces the amount of code in the middleware by removing unnecessary cookie checks, variables, and functions. Instead, checks and conditions are now based on the getToken session. This enables more efficient route handling, such as blocking or allowing access based on whether the user is authenticated and their session role. The overall result is a cleaner and more streamlined middleware.

WellesleyMussolini committed:
refactor: add correct types to JWT and session parameters in auth options

- Set appropriate types for the token and user parameters in the JWT method, and the session parameter in the session method, replacing the previous use of `any` in TypeScript.
- Removed outdated comments related to cookie setting, as cookie handling is no longer necessary with session management now based on OAuth sessions.

WellesleyMussolini committed:
refactor: remove unused cookie authentication functions and variables

- Deleted the handleAuthCookie function, which was setting authentication cookies in the browser for authenticated users, as it's no longer needed.
- Removed the AuthenticationCookie enum, which stored the cookie values for default users and admin users. These values were previously set in the handleAuthCookie function.
- Updated the LogoutModal component by removing the handleLogout prop that relied on the handleLogout function from the utils folder (which deleted authentication cookies). The signOut method from NextAuth is now directly used in the button, eliminating the need for the prop.
- Modified the LayoutWrapper to remove the now unnecessary prop from the LogoutModal, simplifying the authentication logic throughout the application.

WellesleyMussolini committed:
refactor: add types to JWT and session methods and set role to session

WellesleyMussolini committed:
fix: resolve Google authentication bug and update remaining issues

- Fixed a bug related to Google authentication.
- Noted that two issues remain to be addressed.

WellesleyMussolini committed:
chore: no longer necessary cookie auth logic

matheus98rocha commented on src/app/(auth)/utils/auth-options.ts:
```diff
@@ -13,24 +41,27 @@ export const authOptions = {
+      return token;
+    },
+    async session({ session, token }: Session) {
+      console.log(token);
```
remover console.log()

WellesleyMussolini replied:
Foi removido

matheus98rocha commented on src/app/(customer)/(pages)/boardgame/[id]/components/button-rent-game/button-rent-game.component.tsx:
```diff
@@ -1,16 +1,16 @@
 "use client"
 
+import { BoardGame } from "@/app/common/types/boardgame.types";
```
Adicione o net (imports limpos) import

Exemplo: 
react e dependencias primeiro

componentes

types

styles


Imports que vem do mesmo lugar ou das mesmas pastas, crie um index na pasta, exemplo pasta componentes e export todos os itens de lá de dentro.

WellesleyMussolini replied:
Me explica isso na proxima aula.

matheus98rocha commented on src/app/(customer)/(pages)/cart/components/cart-summary/cart-summary.component.tsx:
```diff
@@ -6,7 +6,7 @@ import { useContext } from "@/context/context";
 import { ErrorMessage } from "@/app/components/error-message/error-message.component";
```
Neat import - mesmo cenário do comentado anteriormente.

matheus98rocha commented on src/middleware.ts:
```diff
@@ -1,55 +1,48 @@
+"use server"
```
neat import

matheus98rocha commented on src/middleware.ts:
```diff
@@ -1,55 +1,48 @@
-  // if (pathname === "/" && (!adminCookie || !userCookie)) {
-  //   return NextResponse.redirect(new URL(Pathnames.AUTH, request.url));
-  // }
+  if (authenticated) {
```
Não é uma boa prática if dentro de if, use o &&

matheus98rocha commented on src/middleware.ts:
```diff
@@ -1,55 +1,48 @@
-  //   return NextResponse.redirect(new URL(Pathnames.AUTH, request.url));
-  // }
+  if (authenticated) {
+    if (isNotAdmin && (adminProtectedRoutes.includes(pathname as Pathnames) || pathname === Pathnames.ADMIN_AUTH)) {
```
If statement muito grande dificulta a leitura, coloque toda a condicional em uma constante que resuma qual a verificação esta sendo feita.

Exemplo: const isCondition = isNotAdmin && (adminProtectedRoutes.includes(pathname as Pathnames) || pathname === Pathnames.ADMIN_AUTH

Nome isCondition é um exemplo, o nome da váriavel deve ser descritivo e objetivo

matheus98rocha commented on src/middleware.ts:
Arquivo com muitos comentários, caso eles sejam necessário, explique nos comentários, caso não seja necessário favor remover.

matheus98rocha commented on src/middleware.ts:
```diff
@@ -1,55 +1,48 @@
 
-  // // auth redirection
-  // if ((!userCookie || !adminCookie) && pathname === Pathnames.USER) return NextResponse.redirect(new URL(Pathnames.AUTH, request.url));
+  if (isNotAuthenticated && (pathname === Pathnames.CART || pathname === Pathnames.USER)) {
```
Mesmo cenário explicado no ultimo comentário, if com condicional muito grande, salvar em uma constante.

matheus98rocha commented on src/app/(customer)/layout/wrapper/wrapper.layout.tsx:
```diff
@@ -3,7 +3,6 @@ import "react-toastify/dist/ReactToastify.css";
-import { handleLogout } from "../../utils/handle-logout";
 
 export const LayoutWrapper = () => (
   <>
```
Acredito que esse fragment não é necessário.

matheus98rocha reviewed (commented):
Lembre de ajustar todos os imports usando imports limpos, caso tenha alguma dúvida me avise.

Não se esqueça de remover todos os console.log e todos os comentários desnecessários.

WellesleyMussolini committed:
refactor: simplify middleware logic and clean up code

- Simplified middleware logic by removing nested if statements for authenticated users, replacing them with a single if statement combined with && for additional checks.

- Organized imports across various files and removed unnecessary console logs and comments to enhance code clarity and maintainability.
